activemodel/lib/active_model/attribute_assignment.rb

```diff
@@ -36,6 +36,27 @@ def assign_attributes(new_attributes)
 
     alias attributes= assign_attributes
 
+    # Like `BasicObject#method_missing`, `#attribute_writer_missing` is invoked
+    # when `#assign_attributes` is passed an unknown attribute name.
+    #
+    # By default, `#attribute_writer_missing` raises an UnknownAttributeError.
+    #
+    #   class Rectangle
+    #     include ActiveModel::AttributeAssignment
+    #
+    #     attr_accessor :length, :width
+    #
+    #     def attribute_writer_missing(name, value)
+    #       Rails.logger.warn "Tried to assign to unknown attribute #{name}"
+    #     end
+    #   end
+    #
+    #   rectangle = Rectangle.new
+    #   rectangle.assign_attributes(volume: 10) # => Logs "Tried to assign to unknown attribute 'volume'"
+    def attribute_writer_missing(name, value)
```

Comment:
I don't _love_ the method name.

The PR description outlines prior art, but I'm not sure how to strike a balance between explicit and direct (drawing inspiration from `#method_missing`) and idiomatic (drawing inspiration from `ActiveModel::AttributeMethod`-[generated attribute methods](https://edgeapi.rubyonrails.org/classes/ActiveModel/AttributeMethods.html)).

Reply:
What about `attribute_missing` ? It's straightforward and clear enough to me.

Reply:
`attribute_missing` exists already, but only for readers.